lib/tests/streamlit/proto_compatibility_test.py

```diff
@@ -12,37 +12,206 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+from google.protobuf.descriptor import FieldDescriptor
+from parameterized import parameterized
+
+from streamlit.proto.Alert_pb2 import Alert
+from streamlit.proto.AppPage_pb2 import AppPage
 from streamlit.proto.Common_pb2 import FileURLs, FileURLsRequest, FileURLsResponse
+from streamlit.proto.Exception_pb2 import Exception as Exception_
+from streamlit.proto.NewSession_pb2 import (
+    Config,
+    CustomThemeConfig,
+    EnvironmentInfo,
+    FontFace,
+    FontSizes,
+    Initialize,
+    NewSession,
+    Radii,
+    UserInfo,
+)
+from streamlit.proto.ParentMessage_pb2 import ParentMessage
+from streamlit.proto.SessionStatus_pb2 import SessionStatus
+
+FD = FieldDescriptor
+
+
+@parameterized.expand(
+    [
+        (
+            AppPage,
+            {
+                ("page_script_hash", FD.LABEL_OPTIONAL, FD.TYPE_STRING),
```

Comment:
nit: In case the field is actually annotated with `optional` (which adds the `has..` check method), would it use the same `LABEL_OPTIONAL` label?

Reply:
seems like it -- I couldn't find any obvious way that the field descriptors changed when the `optional` keyword was added